server/src/main/java/org/opensearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -129,11 +130,10 @@ public GlobalOrdinalsStringTermsAggregator(
         this.resultStrategy = resultStrategy.apply(this); // ResultStrategy needs a reference to the Aggregator to do its job.
         this.valuesSource = valuesSource;
         final IndexReader reader = context.searcher().getIndexReader();
-        final SortedSetDocValues values = reader.leaves().size() > 0
+        final SortedSetDocValues values = !reader.leaves().isEmpty()
```

Comment:
Apologies if I am missing something, why we discard `values` here but than again fetch them in `getDocValues()` ? Wouldn't those be the same?

Reply:
@reta Yes they should be the same. However, these are initialized on the `search` thread when the `GlobalOrdinalsStringTermsAggregator` class is initialized but the dvs are used later in `buildAggregation` which we are moving to the `index_searcher` thread in this change.

Reply:
To be clear, these `DocValues` are per Aggregator so there's no concurrent access, Lucene just doesn't expect us to use the data structure in this way where it's created on one thread and passed to another later.